test/hack/update-bicep-json.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+project_root="$(dirname "${BASH_SOURCE}")/../.."
+
+ACTUAL_OUTPUT_DIR="${project_root}/test/e2e/test-artifacts/generated-test-artifacts"
+OUTPUT_DIR="${1:-${ACTUAL_OUTPUT_DIR}}"
+
+# create output directory if it doesn't exist
+mkdir -p "${OUTPUT_DIR}/standard-cluster-create"
+
+# Function to calculate SHA256 hash of a file
+calculate_hash() {
+    local file="$1"
+    sha256sum "$file" | cut -d ' ' -f 1
+}
+
+# Function to store hash in a file
+store_hash() {
+    local hash="$1"
+    local output_file="$2"
+    echo "$hash" > "$output_file"
+}
+
+# Function to check if hash matches stored hash
+check_hash() {
+    local file="$1"
+    local hash_file="${file}.sha256"
+    if [ ! -f "$hash_file" ]; then
+        return 1
+    fi
+    local current_hash=$(calculate_hash "$file")
+    local stored_hash=$(cat "$hash_file")
+    [ "$current_hash" = "$stored_hash" ]
+}
+
+# Generate customer-infra.json if needed
+# Convert bicep file to json and store its hash
+convert_bicep_to_json() {
+    local bicep_file="$1"
+    local json_file="$2"
+
+    if [ ! -f "${json_file}" ] || \
+       ! check_hash "${json_file}" || \
```

Comment:
why do you store the hash of the output? unnecessary

Reply:
If someone changes the output, we need to stomp it